### PR TITLE
PD-22306 spec and po and video turned off

### PR DESCRIPTION
**Pull request by DreamLinks01**
tax calculation for taxable and non taxable product in a same order

### CODE REVIEW

File: integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import { storeFront } from '../../../support/pageObject/storeFront.po';
+const faker = require('faker');
+
+describe("Verify tax for a taxable and non taxable product", () => {
+
+    let qty = 1;
+    let taxRegionTexas = "TX"
+    let taxRateTexas = 0.0825
+    let taxRegionCali = "LA"
+    let taxRateCali = 0.1
```
Comment (bfrishman) [inline integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js:10]: I think these lets can be consts

File: integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                let email = faker.internet.email();
+                let shipFName = faker.name.firstName();
+                let shipLName = faker.name.lastName();
+                let shipState = "Texas"
+                let shipCity = "Houston";
+                let shipZip = "77084";
+
+                let shipStateCali = "California"
+                let shipCityCali = "LA";
+                let shipZipCali = "92509";
```
Comment (bfrishman) [inline integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js:86]: These lets can be consts as well

File: integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                storeFront.getShipPhoneNoInput().scrollIntoView().click()
+                cy.wait(5000);
+
+                cy.log("Verify tax and total with  different shipping methods")
+                storeFront.getUPSGroundShipMethod().check()
+                cy.wait(2000)
+                storeFront.verifyTaxWithTwoProducts(products[0],products[1], taxRegionTexas, taxRateTexas)
+
+                storeFront.getUPS2ndDayAirShipMethod().check()
+                cy.wait(2000)
```
Comment (bfrishman) [inline integration/storeFront/checkoutPage/tax.nonAndTaxableProduct.spec.js:107]: Highly recommend you guys start using semicolons they make code significantly easier to read where the end of lines are.

File: support/pageObject/storeFront.po.js [hunk 4]
```diff
@@ -243,4 +243,39 @@ export class storeFront {
...
+                        } else {
+                            expectedTax = parseFloat(taxProdSubTotal * taxRate).toFixed(2)
+                        }
+                        expect(parseFloat(expectedTax)).to.eq(actualTax)
+                        expect(parseFloat(subTotal)).to.eq(parseFloat(taxProdSubTotal + nonTaxProdSubTotal))
+                        let expectedTotal = parseFloat(taxProdSubTotal + nonTaxProdSubTotal + shipFee + parseFloat(expectedTax)).toFixed(2)
+                        expect(parseFloat(expectedTotal)).to.eq(actualTotal)
+                    })
+                })
+            })
```
[the remark this answers is not in the document]
Comment (bfrishman) [reply support/pageObject/storeFront.po.js:278]: Also more cases where you could use const.

[the remark this answers is not in the document]
Comment (DreamLinks01) [reply support/pageObject/storeFront.po.js:278]: Thanks for all the comments...These are good to have changes. I will have to create a ticket to add all these changes to the entire repo as i just don't want to resolve it for only this spec file. So we can pls go ahead and merge this pr.

Thanks

**Review by dma-cart — APPROVED** [review]

**Pull request merged**